### PR TITLE
vimPlugins.earthly-vim: init at 2024-04-02

### PR DESCRIPTION
**Pull request by DataHearth**
## Description of changes

Add [earthly.vim](https://github.com/earthly/earthly.vim)

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by DataHearth** [comment]
Result of `nixpkgs-review pr 340732` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>vimPluginsUpdater</li>
  </ul>
</details>

**Review by PerchunPak — CHANGES_REQUESTED** [review]
Could you please add an entry to `generated.nix`? Just run `nix-shell -p vimPluginsUpdater --run 'vim-plugins-updater -n'` and make sure to include only the changes related to this plugin. Thanks!

**Comment by DataHearth** [comment]
Hey! Thanks for the review. I though this would be automatic on merge or some CI jobs would do it. I update the file, can you check if this what it's supposed to be inside? First time adding a `vim` plugins ^^". Thanks! @PerchunPak

**Review by PerchunPak — APPROVED** [review]
LGTM

CC @GaetanLepage

**Comment by DataHearth** [comment]
@GaetanLepage done :+1:

**Review by GaetanLepage — APPROVED** [review]
LGTM !

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 340732` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>vimPlugins.earthly-vim</li>
    <li>vimPluginsUpdater</li>
  </ul>
</details>

File: pkgs/applications/editors/vim/plugins/vim-plugin-names [hunk 1]
```diff
@@ -201,8 +201,8 @@ https://github.com/tjdevries/colorbuddy.nvim/,,
 https://github.com/lilydjwg/colorizer/,,
 https://github.com/Domeee/com.cloudedmountain.ide.neovim/,HEAD,
 https://github.com/wincent/command-t/,,
-https://github.com/numtostr/comment.nvim/,,
 https://github.com/LudoPinelli/comment-box.nvim/,HEAD,
+https://github.com/numtostr/comment.nvim/,,
```
[the remark this answers is not in the document]
Comment (PerchunPak) [reply pkgs/applications/editors/vim/plugins/vim-plugin-names]: I got a bit tired of saying this to every PR that adds a plugin...

**Pull request merged**